cpp/src/arrow/memory_pool.h

```diff
@@ -35,44 +36,62 @@ namespace internal {
 ///////////////////////////////////////////////////////////////////////
 // Helper tracking memory statistics
 
-class MemoryPoolStats {
- public:
-  MemoryPoolStats() : bytes_allocated_(0), max_memory_(0) {}
-
-  int64_t max_memory() const { return max_memory_.load(); }
-
-  int64_t bytes_allocated() const { return bytes_allocated_.load(); }
-
-  int64_t total_bytes_allocated() const { return total_allocated_bytes_.load(); }
+class alignas(/*CacheLineSize=*/64) MemoryPoolStats {
```

Comment:
Do you want to follow the pattern below:
```
#ifdef __cpp_lib_hardware_interference_size
    using std::hardware_constructive_interference_size;
    using std::hardware_destructive_interference_size;
#else
    // 64 bytes on x86-64 │ L1_CACHE_BYTES │ L1_CACHE_SHIFT │ __cacheline_aligned │ ...
    constexpr std::size_t hardware_constructive_interference_size = 64;
    constexpr std::size_t hardware_destructive_interference_size = 64;
#endif
```

Reply:
I prefer to fix this on 64-byte and have fixed alignment/sizes of objects (instead of relying on a platform/implementation dependent constant). If a new platform shows up where 64 is not enough and benchmarks prove that a higher alignment is necessary we can consider the size cost against the false-sharing problems and bump it up.